### PR TITLE
Implementation of embedded_hal::digital::IoPin

### PR DESCRIPTION
**Pull request by MorganR**
This is just a draft, and is pending the PR merge of an [IoPin implementation](https://github.com/rust-embedded/embedded-hal/pull/269) into embedded_hal.

I've verified it with a [no-std DHT11 driver based on the IoPin PR](https://github.com/MorganR/rust-simple-sensors/blob/main/src/dht11.rs).

### CODE REVIEW

File: Cargo.toml
Comment (MorganR) [inline Cargo.toml:23]: This needs to be undone before the merge

**Comment by MorganR** [comment]
This has been updated to match the definition of IoPin as merged in [embedded_hal PR 269](https://github.com/rust-embedded/embedded-hal/pull/269).

File: Cargo.toml
Comment (eldruin) [reply Cargo.toml:23]:
```suggestion
embedded-hal = { git = "https://github.com/rust-embedded/embedded-hal" }
```
So that we can run the CI here.

Comment (MorganR) [reply Cargo.toml:23]: Done. Unfortunately this can't build at the moment since the [try_* prefix was recently removed](https://github.com/rust-embedded/embedded-hal/commit/98196625a02dde104197b51c003b08d96fea3207), and this implementation's methods are not compatible with this yet.

**Comment by MorganR** [comment]
Hey folks, I've added a commit to remove the try_ prefixes so this can build against the embedded-hal HEAD. Would you rather merge it like this, or wait until embedded-hal pushes the next alpha release?

**Comment by eldruin** [comment]
Thanks! It would be better only to depend on released versions of `embedded-hal`.

**Comment by eldruin** [comment]
@MorganR Could you update this PR now that I updated this repository to `embedded-hal` `1.0.0-alpha.5`?

**Comment by MorganR** [comment]
Absolutely, I'll update it by the end of the weekend

**Review by eldruin — APPROVED** [review]
Looks good to me, thank you!
bors r+

**Pull request merged**
